DemoApp/Sources/Views/CallView/CallingView/DetailedCallingView.swift

```diff
@@ -8,233 +8,214 @@ import StreamVideoSwiftUI
+                }
+                .padding(.trailing)
+            }
+            .background(Color(appearance.colors.background))
+            .clipShape(RoundedRectangle(cornerRadius: 8))
+            .overlay(RoundedRectangle(cornerRadius: 8).stroke(Color(appearance.colors.textLowEmphasis), lineWidth: 1))
+            .padding(.bottom, 4)
+            .padding(.horizontal)
+
+            Picker("Call action", selection: $callAction) {
+                Text(CallAction.startCall.rawValue).tag(CallAction.startCall)
+                Text(CallAction.joinCall.rawValue).tag(CallAction.joinCall)
+            }
+            .pickerStyle(.segmented)
+            .padding(.horizontal)
+
+            if callAction == .startCall {
+                List(participants) { participant in
+                    Button {
+                        if selectedParticipants.contains(participant) {
+                            selectedParticipants.removeAll { user in
+                                user.id == participant.id
+                            }
+                        } else {
+                            selectedParticipants.append(participant)
+                        }
+                    } label: {
+                        HStack {
+                            Label {
+                                Text(participant.name)
+                                    .frame(maxWidth: .infinity, alignment: .leading)
+                            } icon: {
+                                UserAvatar(imageURL: participant.imageURL, size: imageSize)
+                            }
 
-                    HStack {
-                        Spacer()
-                        ZStack {
-                            if appState.loading {
-                                ProgressView()
-                            } else {
-                                NavigationLink {
-                                    CallsView()
-                                } label: {
-                                    Text("Calls")
-                                }
+                            if selectedParticipants.contains(participant) {
+                                Image(systemName: "checkmark")
                             }
                         }
-                        .padding()
+                        .padding(4)
                     }
+                    .listRowBackground(Color.clear)
+                    .clipShape(RoundedRectangle(cornerRadius: 8))
                 }
-
-                Picker("Call action", selection: $callAction) {
-                    Text(CallAction.startCall.rawValue).tag(CallAction.startCall)
-                    Text(CallAction.joinCall.rawValue).tag(CallAction.joinCall)
+                .listStyle(.plain)
+                .frame(maxWidth: .infinity, alignment: .leading)
+                .accessibility(identifier: "participantList")
+
+                Picker("Call flow", selection: $callFlow) {
```

Comment:
Could this benefit from `CaseIterable` and `ForEach` (what I was mentioning before)?